src/config.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import type {
+  OAuthConfig,
+  OAuthPredefinedConfig,
+  OAuthProviderInfo,
+  OAuthUserConfig,
+} from "./types.ts";
+import { assert } from "https://deno.land/std@0.201.0/assert/assert.ts";
+
+export function createOAuthConfig(
```

Comment:
I should probably jsdoc this!

But this adds consistency to the providers by combining the hardcoded config with the env vars and user supplied config in a consistent manner.